metrics/metrics.go

```diff
@@ -29,7 +29,8 @@ type Config struct {
 type Metricer interface {
 	RecordInfo(version string)
 	RecordUp()
-	RecordRPCServerRequest(method string) func(status string)
+	RecordBadRequestHeader(method string, errorType string)
```

Comment:
technically the commitment is passed via a URL parameter and not a request header - tangentially we're already recording response status codes which should encapsulate this metric. Is there any reason to have this?

Reply:
yep, i was thinking about this too. I've combined the new metrics to `RecordRPCServerRequest` and record commitment mode and version (or errors of reading them) after handling the request; I modified `handleFunc` to read the schemes and return them after handling so the work isn't doubled, and the reason to have `handleFunc` to return them instead of reading them in is so that the server can easily write bad requests response to the caller